### PR TITLE
Add ES-Reader GetTrace

### PR DESCRIPTION
**Pull request by mh-park**

### CODE REVIEW

**Comment by black-adder** [comment]
looks like you need to run a make fmt

File: model/converter/json/to_domain.go [hunk 1]
```diff
@@ -184,6 +185,9 @@ func (td toDomain) convertLogs(logs []json.Log) ([]model.Log, error) {
 }
 
 func (td toDomain) convertProcess(process *json.Process) (*model.Process, error) {
+	if process == nil {
```
Comment (black-adder) [inline model/converter/json/to_domain.go]: what warranted this? how could the process be missing?

File: plugin/storage/es/spanstore/reader.go [hunk 2]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
 
 // GetTrace takes a traceID and returns a Trace associated with that traceID
 func (s *SpanReader) GetTrace(traceID model.TraceID) (*model.Trace, error) {
-	// TODO
-	return nil, nil
+	return s.readTrace(traceID.String(), spanstore.TraceQueryParameters{})
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: dont think readTrace adds any value, can inline here

File: model/converter/json/to_domain_test.go [hunk 3]
```diff
@@ -247,4 +255,4 @@ func TestFailureBadParentSpanID(t *testing.T) {
 	require.NoError(t, err)
 	badParentSpanIDESSpan.ParentSpanID = "zz"
 	failingSpanTransformAnyMsg(t, &badParentSpanIDESSpan)
-}
+}
```
Comment (black-adder) [inline model/converter/json/to_domain_test.go]: are you manually removing the empty lines from the end of the files?

File: plugin/storage/es/spanstore/reader.go [hunk 4]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+}
+
+func (s *SpanReader) readTrace(traceID string, traceQuery spanstore.TraceQueryParameters) (*model.Trace, error) {
+	query := elastic.NewTermQuery("traceID", traceID)
+
+	indices := s.findIndices(traceQuery)
+	esSpansRaw, err := s.executeQuery(query, indices...)
+	if err != nil {
+		return nil, err
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: wrap error

File: plugin/storage/es/spanstore/reader.go [hunk 5]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+	}
+	if len(esSpansRaw) == 0 {
+		return nil, spanstore.ErrTraceNotFound
+	}
+
+	spans := make([]*model.Span, len(esSpansRaw))
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: L81 to L93 can be it's own function

File: plugin/storage/es/spanstore/reader.go [hunk 6]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+
+	for i, esSpanRaw := range esSpansRaw {
+		jsonSpan, err := s.esJSONtoJSONSpanModel(esSpanRaw)
+		if err != nil {
+			return nil, err
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: wrap error

File: plugin/storage/es/spanstore/reader.go [hunk 7]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+		}
+		span, err := jConverter.SpanToDomain(jsonSpan)
+		if err != nil {
+			return nil, err
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: ditto

File: plugin/storage/es/spanstore/reader.go [hunk 8]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+		}
+		spans[i] = span
+	}
+
+	trace := &model.Trace{}
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: you can reduce these 3 lines to 1

File: plugin/storage/es/spanstore/reader.go [hunk 9]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+		Type(spanType).
+		Query(query).
+		Do(s.ctx)
+	if err != nil {
+		return nil, err
+	}
+	return searchService.Hits.Hits, nil
+}
+
+func (s *SpanReader) esJSONtoJSONSpanModel(esSpanRaw *elastic.SearchHit) (*jModel.Span, error) {
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: rename to unmarshallJSONSpan

File: plugin/storage/es/spanstore/reader.go [hunk 10]
```diff
@@ -60,8 +63,60 @@ func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
...
+	esSpanInByteArray := esSpanRaw.Source
+
+	var jsonSpan jModel.Span
+	err := json.Unmarshal(*esSpanInByteArray, &jsonSpan)
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: inline with below

File: plugin/storage/es/spanstore/reader_test.go [hunk 11]
```diff
@@ -38,6 +38,45 @@ import (
 	"github.com/uber/jaeger/storage/spanstore"
 )
 
+var exampleESSpan = []byte(`{      "traceID": "1",
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader_test.go]: can this be formatted a bit better? there's too much indentation, start the json blob on a new line

File: model/converter/json/to_domain_test.go (hunk 3 above)
Comment (mh-park) [inline model/converter/json/to_domain_test.go]: no make fmt is doing that

Comment (mh-park) [inline model/converter/json/to_domain_test.go]: or something

File: plugin/storage/es/spanstore/reader.go (hunk 2 above)
Comment (mh-park) [inline plugin/storage/es/spanstore/reader.go]: i did this to follow cassandra syntax + i didn't want to call `GetTrace` in `FindTraces`

File: model/converter/json/to_domain.go (hunk 1 above)
Comment (mh-park) [inline model/converter/json/to_domain.go]: don't think we ever will, but if we convert UI model to domain process will be nil

File: plugin/storage/es/spanstore/reader.go (hunk 2 above)
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: good point, we'll need readTrace in FindTraces

File: model/converter/json/to_domain.go (hunk 1 above)
Comment (mh-park) [inline model/converter/json/to_domain.go]: fyi it happened when i was running a basic test with a json model span with just traceID bc i didnt need all the parts

**Pull request merged**